### DISCUSSION TITLE
Extending binary operators in logical plans

### DISCUSSION THREAD

**Discussion by renzobautista** [apache/datafusion#11956] [Q&A] [answered]
Hi all, I'm currently using DataFusion to generate logical plans from a user-inputted SQL query. In my custom SQL implementation, I want to allow some binary operators (+,-,*,/) to take in strings as input (i.e. coercing Utf8 + Utf8 to Float64). Is this currently possible? I was able to customize other scalar functions and aggregations by making a UDF (e.g. for SUM, CEIL), but haven't found a way to do the same for binary operators.

**Comment by jonahgao** [comment, marked as answer]
Can customizing SQL planner meet your requirement? Here is an [example](https://github.com/apache/datafusion/blob/f98f8a92128451614d334db731c61682d134db6f/datafusion/core/tests/user_defined/expr_planner.rs#L34).

**Reply by renzobautista** [reply]
i think this should work, will give it a go -- thanks @jonahgao !
